File: archives/python/11723.py
import sys
input = sys.stdin.readline

M = int(input())
S = set()
for _ in range(M):
	command = input().split()
	if command[0] == "add":
		x = int(command[1])
		S.add(x)
	elif command[0] == "remove":
		x = int(command[1])
		if x in S:
			S.remove(x)
		else:
			continue
	elif command[0] == "check":
		x = int(command[1])
		if x in S:
			print(1)
		else:
			print(0)
	elif command[0] == "toggle":
		x = int(command[1])
		if x in S:
			S.remove(x)
		else:
			S.add(x)
	elif command[0] == "all":
		S = set([1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
				11, 12, 13, 14, 15, 16, 17, 18, 19, 20])
	elif command[0] == "empty":
		S = set()

File: archives/cpp/11723.cpp
#include <bits/stdc++.h>
#define FOR(i, start, end) for (int i = start; i < end; i++)
#define endl "\n"
#define fastio ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
#pragma GCC optimize("Ofast")
using namespace std;
// set으로 연산하면 시간초과남.
int arr[21];

int main() {
    fastio

    int M; cin >> M;
    int temp;
    while (M--) {
        string inst; cin >> inst;
        if (inst == "add") {
            cin >> temp;
            arr[temp] = 1;
        } else if (inst == "remove") {
            cin >> temp;
            arr[temp] = 0;
        } else if (inst == "check") {
            cin >> temp;
            cout << (arr[temp] == 1 ? 1 : 0) << endl;
        } else if (inst == "toggle") {
            cin >> temp;
            arr[temp] = abs(arr[temp]-1);
        } else if (inst == "all") {
            fill(arr+1, arr+21, 1);
        } else if (inst == "empty") {
            fill(arr+1, arr+21, 0);
        }
    }

    return 0;
}